src/sources/soundsource.cpp

```diff
@@ -48,7 +48,8 @@ QString SoundSource::getTypeFromFile(const QFileInfo& fileInfo) {
     const QString fileSuffix = fileInfo.suffix();
     const QString fileType = fileTypeFromSuffix(fileSuffix);
     DEBUG_ASSERT(!fileType.isEmpty() || fileType == QString{});
-    const QMimeType mimeType = QMimeDatabase().mimeTypeForFile(fileInfo);
+    const QMimeType mimeType = QMimeDatabase().mimeTypeForFile(
+            fileInfo, QMimeDatabase::MatchContent);
```

Comment:
The control flow of this function is hard to understand. Along all the checks, it is hard to see what will take priority. 
I think it should be: 
1: MatchContent 
2: MatchExtension
3: If we think we are smarter than MatchExtension use our fileType, which is essential just the suffix. (Maybe we can even drop it. It is just a hack around the "aif" issue, that might be already handled in QMimeDatabase 

Is it correct, that may follows this pseudo code? 

```
const QMimeType mimeType = QMimeDatabase().mimeTypeForFile(
        fileInfo, QMimeDatabase::MatchContent);
if(mimeType.isDefault()) {
    mimeType = QMimeDatabase().mimeTypeForFile(
            fileInfo, QMimeDatabase::MatchExtension);
    if(mimeType.isDefault()) {
        mimeType = QMimeDatabase().mimeTypeForFile(
                fileInfo, QMimeDatabase::MatchExtension);
    }
}
const QString preferredSuffix = mimeType.preferredSuffix();
if (!preferredSuffix.isEmpty()) {
    return prefferedSuffix;
}
const QString fileSuffix = fileInfo.suffix();
retun preferredSuffixFromSuffix(fileSuffix);  // old fileTypeFromSuffix
```

Reply:
By the way I like the name "preferredSuffix" much more the fileType. It is hard to see what out propritary term "fileTtype" really is and the sound sources converts it back to suffix anyway any conversion function.

Reply:
I have just verified, we can drop our custom aif aiff hack. QMimeDatabase::MatchExtension does it for us.

Reply:
I suppose the method can be refactored in a follow-up PR. Initially, I only wanted to fix the temp file issue with our tests. The only reason why I touched this file at all in this PR is because fixing the leftover temp files revealed that the temp file names were also wrong, which in turn revealed a bug in our file type detection code (the test fails when the intended filenames are used).

Reply:
The distinction was necessary because each "file type" in Mixxx (e.g. "aiff") may be associated with multiple "file suffixes". This originated from the wrong assumption of a 1:1 relation between file types and file suffixes before.

Reply:
@daschuer I agree with Jan, please do not extend the scope of this PR!

Reply:
My proposal was just to rename "fileType" to "preferredSuffix" and rely on QMimeType.

Alternatively we may replace all FileTyp instances with QMimeTye and look up the preferredSuffix whenever needed.

What did you think?